parse/analyze.go

```diff
@@ -426,21 +426,21 @@ var _ Visitor = (*sqlAnalyzer)(nil)
 // typeErr should be used when a type error is encountered.
 // It returns an unknown attribute and adds an error to the error listener.
 func (s *sqlAnalyzer) typeErr(node Node, t1, t2 *types.DataType) *types.DataType {
-	s.errs.AddErr(node, ErrType, fmt.Sprintf("%s != %s", t1.String(), t2.String()))
+	s.errs.AddErr(node, ErrType, "%s != %s", t1, t2)
```

Comment:
There are three "*Err" functions with this `msg string, args ...any` pattern.  Sometimes used, sometimes not.  Honestly it would be better with just `msg string` leaving the caller to format if desired, as was the case in some places but not others, but I am leaving the function signatures unchanged for now.